torch/distributed/fsdp/flatten_params_wrapper.py

```diff
@@ -167,45 +190,98 @@ def __init__(self, params: Sequence[nn.Parameter], requires_grad: bool = True):
         # The indices (begin/end pair) of the parameters that are included in
         # this FlatParameter. The default value is all the parameters because
         # no sharding happen yet.
-        self._param_indice_in_shard = (0, len(self._param_infos) - 1)
+        self._param_indice_in_shard = Interval(0, len(self._param_infos) - 1)
```

Comment:
seems ParamOffset is an alias for Interval, do we want to just use one or the other (i.e. change this to ParamOffset)?